packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useUploadFileEntities.ts

```diff
@@ -350,13 +355,15 @@ export function useUploadFileEntities(
     [postPrepareUpload, prepareUpload],
   )
 
-  const activePrompts = useMemo(() => {
+  const activePrompts: Prompt[] = useMemo(() => {
     return filesToConfirmNewVersion.map(fileToPrompt => {
       return {
-        title: 'Update existing file?',
-        message: `A file named "${fileToPrompt.file.name}" (${
-          (fileToPrompt as UpdateEntityFileUpload).existingEntityId
-        }) already exists in this location. Do you want to update the existing file and create a new version?`,
+        info: {
+          type: 'CONFIRM_NEW_VERSION',
+          fileName: fileToPrompt.file.name,
+          existingEntityId: (fileToPrompt as UpdateEntityFileUpload)
+            .existingEntityId,
+        },
```

Comment:
Refactored so this hook just provides the information. The component will transform it into user-friendly text.